reporter/cloudwatch.go

```diff
@@ -102,53 +104,64 @@ func metricsData(cfg *config.Config) []*cloudwatch.MetricDatum {
 			// We don't clear gauge counters on ResetCountersOnReport because they cannot recover their value like normal gauges
 			// They can only increment and decrement and so cannot be cleared after they are created.
 		case metrics.Gauge:
-			gagues += 1
+			gauges += 1
 			value := float64(metric.Value())
 			if cfg.Filter.ShouldReport(name, value) {
 				datum := aDatum(name)
 				datum.Unit = aws.String(cloudwatch.StandardUnitCount)
 				datum.Value = aws.Float64(float64(value))
 				data = append(data, datum)
-				gaguesOut += 1
+				gaugesOut += 1
 			}
 		case metrics.GaugeFloat64:
-			gagues += 1
+			gauges += 1
 			value := float64(metric.Value())
 			if cfg.Filter.ShouldReport(name, value) {
 				datum := aDatum(name)
 				datum.Unit = aws.String(cloudwatch.StandardUnitCount)
 				datum.Value = aws.Float64(value)
 				data = append(data, datum)
-				gaguesOut += 1
+				gaugesOut += 1
 			}
 		case metrics.Histogram:
 			histos += 1
 			h := metric.Clear() // For the LD flavor of go-metrics we do an atomic snapshot and clear
-			value := float64(h.Count())
-			if cfg.Filter.ShouldReport(name, value) {
-				for _, p := range cfg.Filter.Percentiles(name) {
-					pname := fmt.Sprintf("%s-perc%.3f", name, p)
-					pvalue := h.Percentile(p)
-					if cfg.Filter.ShouldReport(pname, pvalue) {
-						datum := aDatum(pname)
-						datum.Unit = aws.String(cloudwatch.StandardUnitCount)
-						datum.Value = aws.Float64(pvalue)
-						data = append(data, datum)
-						histosOut += 1
-					}
+
+			for n, v := range map[string]float64{
+				fmt.Sprintf("%s.count", name):   float64(h.Count()),
+				fmt.Sprintf("%s.min", name):     float64(h.Min()),
+				fmt.Sprintf("%s.max", name):     float64(h.Max()),
+				fmt.Sprintf("%s.mean", name):    h.Mean(),
+				fmt.Sprintf("%s.std-dev", name): h.StdDev(),
+			} {
+				if cfg.Filter.ShouldReport(n, v) {
+					datum := aDatum(n)
+					datum.Value = aws.Float64(v)
+					data = append(data, datum)
+					histosOut += 1
+				}
+			}
+			for _, p := range cfg.Filter.Percentiles(name) {
+				percentileLabel := strings.Replace(strconv.FormatFloat(p*100.0, 'f', -1, 64), ".", "", 1)
```

Comment:
I think this makes sense but could you maybe put an example input and output in a comment? :)